### PR TITLE
 enable k3s cloud provider if node-external-ip flag is passed to rke2 

### PR DESCRIPTION
**Pull request by MonzElmasry**
#### Linked Issues ####

https://github.com/rancher/rke2/issues/309

### CODE REVIEW

**Comment by MonzElmasry** [comment]
progress is blocked by https://github.com/rancher/k3s/issues/2345

File: Dockerfile [hunk 1]
```diff
@@ -117,6 +117,9 @@ RUN CHART_VERSION="v1.18.10"     CHART_FILE=/charts/rke2-kube-proxy.yaml
 RUN CHART_VERSION="2.11.100"    CHART_FILE=/charts/rke2-metrics-server.yaml    CHART_BOOTSTRAP=false   /charts/build-chart.sh
 RUN rm -vf /charts/*.sh /charts/*.md
 
+FROM manifests AS manifests
+COPY manifests/ccm.yaml /manifests/
```
Comment (brandond) [inline Dockerfile:121]: We are not packaging roles and such as manifest files for rke2. The only thing we deploy should be helm charts. If you need to add RBAC it should be added in here somewhere: https://github.com/rancher/rke2/blob/master/pkg/rke2/clusterrole_templates.go

**Review by brandond — CHANGES_REQUESTED** [review]

**Review by dweomer — APPROVED** [review]

**Review by brandond — APPROVED** [review]

**Comment by fentas** [comment]
Thanks :+1: 
Documentation for this is still missing, though.

**Comment by brandond** [comment]
@fentas what are you looking for documentation on?

**Comment by fentas** [comment]
https://docs.rke2.io/install/install_options/server_config/

`node-external-ip` is not documented here as a parameter. : )

**Pull request merged**